File: server/datasets/java_source-code/Multiple-D1/organic_7.java
/*
 * Quick Sort - Organic Submission #7
 * Functional-style partitioned List Quick Sort.
 */

import java.util.*;

class Solution {
    public static List<Integer> quickSortList(List<Integer> list) {
        if (list.size() <= 1) return list;
        int pivot = list.get(list.size() / 2);
        List<Integer> left = new ArrayList<>();
        List<Integer> middle = new ArrayList<>();
        List<Integer> right = new ArrayList<>();

        for (int x : list) {
            if (x < pivot) left.add(x);
            else if (x == pivot) middle.add(x);
            else right.add(x);
        }

        List<Integer> result = new ArrayList<>();
        result.addAll(quickSortList(left));
        result.addAll(middle);
        result.addAll(quickSortList(right));
        return result;
    }

    public static void main(String[] args) {
        List<Integer> nums = Arrays.asList(33, 10, 55, 71, 29, 62);
        System.out.println(quickSortList(nums));
    }
}

File: server/datasets/python_source-code/Multiple-D1/organic_7.py
"""
Quick Sort - Organic Submission #7
Functional-style list comprehension Quick Sort.
"""

def quick_sort_pythonic(arr):
    if len(arr) <= 1:
        return arr
    pivot = arr[len(arr) // 2]
    left = [x for x in arr if x < pivot]
    middle = [x for x in arr if x == pivot]
    right = [x for x in arr if x > pivot]
    return quick_sort_pythonic(left) + middle + quick_sort_pythonic(right)

if __name__ == "__main__":
    nums = [33, 10, 55, 71, 29, 62]
    print(quick_sort_pythonic(nums))
